src/Utils.php

```diff
@@ -45,21 +45,15 @@ public static function calculateBitmask(array $options)
     }
 
     /**
-     * Converts a camel cased string to an snake cased string.
+     * Converts a camel cased string to a snake cased string.
      *
      * @param string $string
      *
      * @return string
      */
     public static function camelCaseToUnderscore($string)
     {
-        return Preg::replaceCallback(
-            '/(^|[a-z0-9])([A-Z])/',
-            static function (array $matches) {
-                return strtolower('' !== $matches[1] ? $matches[1].'_'.$matches[2] : $matches[2]);
-            },
-            $string
-        );
+        return strtolower(Preg::replace('/(?<!^)(?=[A-Z])/', '_', $string));
```

Comment:
This regular expression replaces `AA` with `A_A` (and results in `a_a`).
```suggestion
        return strtolower(Preg::replace('/(?<=[a-z0-9])(?=[A-Z])/', '_', $string));
```
Please add some test cases.

Reply:
Is that wrong? Should be `aa`?

Because there is no test for such case and the previous version is impossible to test for `AA` - the first call changes the string to `aA` and then second to `a_a` - and test checks the transformation and also if the result is not transformed again.

Reply:
I'd say both implementations are wrong. Maybe we could rather remove this method entirely and give `symfony/string` (or another similar library) a try?

Reply:
I wish we can use `symfony/string`, but... PHP 5.6...

I'd updated the RegEx and added few tests (some of them failing on the version with callback) - I've planned to make the function one-liner inside of `Utils` - to show it ~still~ works fine and next propose to remove it from `Utils` and use the new version directly in code - it is used 3 times in the repo.